src/heroes/hanzo/stormbow.opy

```diff
@@ -0,0 +1,24 @@
+#!mainFile "../../dev_main.opy"
+
+playervar stormbow_distance
+
+
+rule "[hanzo/stormbow.opy]: Increase Storm Bow/Sonic Arrow damage in effective range":
+    @Event playerDealtDamage
+    @Hero hanzo
+    @Condition eventAbility in [Button.PRIMARY_FIRE, Button.ABILITY_1]
+
+    # this code makes me want to cry but it works
+    eventPlayer.stormbow_distance = distance(attacker.getEyePosition(), centerMass(victim)) - 0.25
+    if eventPlayer.stormbow_distance <= ADJ_HANZO_DAMAGE_FALLOFF_START_DISTANCE:
+        damage(victim, attacker, ((eventDamage/eventPlayer._base_damage_scalar) * (ADJ_HANZO_ARROW_DAMAGE / OW2_HANZO_ARROW_DAMAGE) - eventDamage)/eventPlayer._base_damage_scalar)
+    else if eventPlayer.stormbow_distance <= ADJ_HANZO_DAMAGE_FALLOFF_END_DISTANCE:
+        damage(victim, attacker, ((eventDamage/eventPlayer._base_damage_scalar) * (ADJ_HANZO_ARROW_DAMAGE / OW2_HANZO_ARROW_DAMAGE) - eventDamage)/eventPlayer._base_damage_scalar * ((ADJ_HANZO_DAMAGE_FALLOFF_END_DISTANCE-eventPlayer.stormbow_distance)/(ADJ_HANZO_DAMAGE_FALLOFF_END_DISTANCE-ADJ_HANZO_DAMAGE_FALLOFF_START_DISTANCE)))
```

Comment:
I need to check if this scales properly with damage boosts and headshots

Reply:
headshot works, but haven't tested damage boost like mercy beam or discord. My guess is that it will work, but I'm gonna rewrite the formula with explanation of how the "damage compensation" tech works